projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -626,6 +626,15 @@ export class IgxTreeGridComponent extends IgxGridBaseDirective implements GridTy
         return this._pinnedRecordIDs.indexOf(id) !== -1;
     }
 
+    /**
+     * @hidden
+     * @internal
+     */
+    public pinRecordIndex(rec) {
+        const id = this.primaryKey ? rec.data[this.primaryKey] : rec.data;
```

Comment:
Could be simplified since you just want to extract the data and run the same code as in base:
public pinRecordIndex(rec) {
        return super.pinRecordIndex(rec.data);
  }